builtin/providers/azure/resource_azure_instance.go

```diff
@@ -647,7 +651,7 @@ func retrieveOSImageDetails(
 			if img.MediaLink == "" {
 				if storage == "" {
 					return nil, "", nil,
-						fmt.Errorf("When using a platform image, the 'storage' parameter is required")
+						PlatformStorageError
```

Comment:
Extremely small nit, but since the line is now way shorter, there's not much reason to place `PlatformStorageError` on it's own line... Through is doesn't hurt either of course :wink: